### PR TITLE
Paginate fetching all assets

### PR DESCRIPTION
**Pull request by salazarm**
## Summary & Motivation

Paginate fetching asset data in batches of 100k

## How I Tested These Changes
Existing jest tests

### CODE REVIEW

File: js_modules/dagster-ui/packages/ui-core/src/assets/AssetsCatalogTable.tsx [hunk 1]
```diff
@@ -26,46 +26,96 @@ import {FIFTEEN_SECONDS, useRefreshAtInterval} from '../app/QueryRefresh';
...
+import {fetchPaginatedData} from '../runs/fetchPaginatedBucketData';
+import {CacheManager} from '../search/useIndexedDBCachedQuery';
 import {LoadingSpinner} from '../ui/Loading';
 
 type Asset = AssetTableFragment;
 
 const groupTableCache = new Map();
 
+const ASSET_CATALOG_TABLE_QUERY_VERSION = 1;
+const BATCH_LIMIT = 100000;
```
Comment (salazarm) [inline js_modules/dagster-ui/packages/ui-core/src/assets/AssetsCatalogTable.tsx:41]: We can lower this limit once we take the limit into account on the backend, but for now 100k seems to work for January-capital

Comment (prha) [reply js_modules/dagster-ui/packages/ui-core/src/assets/AssetsCatalogTable.tsx:41]: We do take the limit into account... I think ideally this batch limit is something like 1000?

Comment (prha) [reply js_modules/dagster-ui/packages/ui-core/src/assets/AssetsCatalogTable.tsx:41]: oh you're right, we only do in OSS.  I can put up a PR

**Comment by prha** [comment]
Can we reuse this pattern for the `SearchSecondaryQuery` also?

Doesn't have to be this PR.

**Comment by salazarm** [comment]
@prha yes

File: js_modules/dagster-ui/packages/ui-core/src/assets/AssetsCatalogTable.tsx [hunk 2]
```diff
@@ -26,46 +26,96 @@ import {FIFTEEN_SECONDS, useRefreshAtInterval} from '../app/QueryRefresh';
...
+          assets: data,
+        });
+      }
+    });
+  }, [cacheManager, assetsRef]);
+
+  const fetchAssets = useCallback(async () => {
+    try {
+      const data = await fetchPaginatedData({
+        async fetchData(cursor: string | undefined) {
```
Comment (bengotow) [inline js_modules/dagster-ui/packages/ui-core/src/assets/AssetsCatalogTable.tsx:73]: Not urgent but while we're in here, could we make fetchData not use  `dataSoFar.push(...data)`? Technically there's a limit to the number of objects we can fetch at a time if we use the spread operator to pass them each as an argument. In JavaScriptCore the limit is `65536` but I think it's higher in Chrome?

If our page size is 100k we could def get close to the max there

**Review by bengotow — APPROVED** [review]
LGTM!

File: js_modules/dagster-ui/packages/ui-core/src/assets/AssetsCatalogTable.tsx [hunk 3]
```diff
@@ -26,46 +26,96 @@ import {FIFTEEN_SECONDS, useRefreshAtInterval} from '../app/QueryRefresh';
...
+          error: undefined,
+          assets: data,
+        });
+      }
+    });
+  }, [cacheManager, assetsRef]);
+
+  const fetchAssets = useCallback(async () => {
+    try {
+      const data = await fetchPaginatedData({
```
Comment (prha) [inline js_modules/dagster-ui/packages/ui-core/src/assets/AssetsCatalogTable.tsx:72]: neat, I didn't realize we already had this.

File: js_modules/dagster-ui/packages/ui-core/src/assets/AssetsCatalogTable.tsx (hunk 2 above)
Comment (bengotow) [reply js_modules/dagster-ui/packages/ui-core/src/assets/AssetsCatalogTable.tsx:73]: edit: Actually it is urgent, turns out you can't `array.push(...100k items)` in Safari at all

image

Comment (salazarm) [reply js_modules/dagster-ui/packages/ui-core/src/assets/AssetsCatalogTable.tsx:73]: Ah right I remember we ran into this before.

Comment (salazarm) [reply js_modules/dagster-ui/packages/ui-core/src/assets/AssetsCatalogTable.tsx:73]: Though we're lowering the limit to 1,000 now 😬

File: js_modules/dagster-ui/packages/ui-core/src/assets/__fixtures__/AssetTables.fixtures.ts [hunk 4]
```diff
@@ -310,6 +310,7 @@ export const AssetCatalogTableMock = buildQueryMock<
   AssetCatalogTableQueryVariables
 >({
   query: ASSET_CATALOG_TABLE_QUERY,
+  variableMatcher: () => true,
```
Comment (prha) [inline js_modules/dagster-ui/packages/ui-core/src/assets/__fixtures__/AssetTables.fixtures.ts:313]: Is there a way to test this with a low batch size, just to make sure that we're getting the pagination logic correctly, without wildcarding the variables?

**Review by prha — APPROVED** [review]

File: js_modules/dagster-ui/packages/ui-core/src/assets/__fixtures__/AssetTables.fixtures.ts (hunk 4 above)
Comment (salazarm) [reply js_modules/dagster-ui/packages/ui-core/src/assets/__fixtures__/AssetTables.fixtures.ts:313]: I'll add a test case for this

**Pull request merged**